src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/ActivitySourceConfiguration.cs

```diff
@@ -0,0 +1,124 @@
+                                }
+
+                                if (!cancellationTokenSource.IsCancellationRequested)
+                                {
+                                    // Note: Version should be the first message
+                                    // written so cancel once we have received a
+                                    // message.
+                                    cancellationTokenSource.Cancel();
+                                }
+                            }
+                            catch (Exception)
+                            {
+                            }
+                        };
+
+                        using EventTaskSource<Action> sourceCompletedTaskSource = new(
+                            taskComplete => taskComplete,
+                            handler => eventSource.Completed += handler,
+                            handler => eventSource.Completed -= handler,
+                            token);
+
+                        await sourceCompletedTaskSource.Task.ConfigureAwait(false);
+                    });
+
+                try
+                {
+                    processor.Process(client, TimeSpan.FromSeconds(10), resumeRuntime: false, token: cancellationTokenSource.Token).GetAwaiter().GetResult();
+                }
+                catch (OperationCanceledException)
+                {
+                }
+
+                processor.DisposeAsync().AsTask().GetAwaiter().GetResult();
+
+                if (majorVersion < 9)
+                {
+                    throw new NotSupportedException("Sampling ratio can only be set when listening to processes running System.Diagnostics.DiagnosticSource 9 or greater");
+                }
+            }
+        }
+
+        public override IList<EventPipeProvider> GetProviders()
+        {
+            StringBuilder filterAndPayloadSpecs = new();
+            foreach (string activitySource in _ActivitySourceNames)
+            {
+                if (string.IsNullOrEmpty(activitySource))
+                {
+                    continue;
+                }
+
+                // Note: It isn't currently possible to get Events or Links off
+                // of Activity using this mechanism:
+                // Events=Events.*Enumerate;Links=Links.*Enumerate; See:
+                // https://github.com/dotnet/runtime/issues/102924
+
+                string sampler = string.Empty;
+
+                if (_SamplingRatio < 1D)
+                {
+                    sampler = $"-ParentRatioSampler({_SamplingRatio})";
+                }
+
+                filterAndPayloadSpecs.AppendLine($"[AS]{activitySource}/Stop{sampler}:-TraceId;SpanId;ParentSpanId;ActivityTraceFlags;TraceStateString;Kind;DisplayName;StartTimeTicks=StartTimeUtc.Ticks;DurationTicks=Duration.Ticks;Status;StatusDescription;Tags=TagObjects.*Enumerate;ActivitySourceVersion=Source.Version");
```

Comment:
Can you add a comment here regarding the fact that DiagnosticSourceEventSource has only 1 possible configuration? The implications of that are that if we ever support multiple configurations at the same time, they must be cumulative. See https://github.com/dotnet/diagnostics/pull/3425